### ISSUE TITLE
How to use graphPost to create a chat? Possible?

### ISSUE THREAD

**Issue by trillian74** [pnp/pnpjs#2736]
I have a question about doing graph posts against chats and how to initialize it?
given this snippet
```
   const graphPostResult = await graphPost(GraphQueryable(graph,"chats"), {
                body: JSON.stringify({
                    "chatType": "oneOnOne",
                    "members": [
                        {
```
graph is giving me this exception: _Argument of type 'GraphFI' is not assignable to parameter of type 'GraphInit'.
  Type 'GraphFI' is missing the following properties from type 'IGraphQueryable<any>': parentUrl, select, expand, getParent, and 21 more_

Is this possible or not supported?

All the best, Ole
### Category
- [ ] Enhancement
- [ ] Bug
- [X ] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [      latest   ]

Please specify what version(s) of SharePoint you are targeting: [     latest    ]

**Comment by trillian74** [comment]
Thank you for helping out. I don’t have it in front of me, but until I do I
use a “standard” pnpConfig.is file and are initiating the sp and graf
before rendering the webpart.

I know this is not much, but if it does than great if not then I’ll update
this tomorrow

Ole

tir. 11. jul. 2023 kl. 17:55 skrev Julie Turner ***@***.***>:

**Comment by juliemturner** [comment]
Just guessing I'd say you're passing "graph" in to SPQueryable but "chat" is probably an endpoint on user... so you need to start with getting the user and then using that queryable instance to pass into GraphQueryable.

**Comment by trillian74** [comment]
Thanks @juliemturner for getting back to me. Appreciate it :)

Just want to add the details:
The graph context
`import { ConsistencyLevel, SPFx as GSPFx, GraphFI, graphfi } from '@pnp/graph';`


`let _graph: GraphFI = null;`
`export const getGraph = (context?: WebPartContext): GraphFI => {`
`	if (context != null) {`
`		_graph = graphfi().using(GSPFx(context)).using(PnPLogging(LogLevel.Warning));`
`	}`
`	return _graph;`
`};
`

So basically I want to create a oneOnOne chat and that needs to be created before sending the message and it all is done thru the endpoint [chats](https://learn.microsoft.com/en-us/graph/api/chat-post?view=graph-rest-1.0&tabs=http) 

POST https://graph.microsoft.com/v1.0/chats

I then thought that graphPost could help me by using graphPost(GraphQueryable(graph,"chats"), {, but then i received the message from the pack saying  **Argument of type 'GraphFI' is not assignable to parameter of type 'GraphInit'.
Type 'GraphFI' is missing the following properties from type 'IGraphQueryable': parentUrl, select, expand, getParent, and 21 more**

But, maybe i should try using the me route... i thought that i needed to create the chats in the chats route... i'll try that first and respond

**Comment by juliemturner** [comment]
Going to close this as addressed in PR #2737. If you have any additional questions please feel free to open a new issue.
